### DISCUSSION TITLE
Bug while drawing

### DISCUSSION THREAD

**Discussion by Agnyy** [opentoonz/opentoonz#6804] [General]
This is not the first time I have been writing about this problem and asking for help in fixing it. This problem manifests itself not only with me.

Screenshot or Video Reference

![image](https://user-images.githubusercontent.com/25137502/78812896-d2823e00-79d4-11ea-81b0-2b7b1a8bc27c.png)

![image](https://user-images.githubusercontent.com/25137502/78812904-d9a94c00-79d4-11ea-9ed2-2a4697c0b6a4.png)

System Information
**OpenToonz Version - nightly build
**Operating System - Windows 10
**CPU - i7 - 6700hq
**Memory - 16
**Graphics Card - GTX 970M

Here is a video of how an error appears.
https://youtu.be/8cM8OQ5uM5w

I have already completely reinstalled the program several times. I tried with different versions of the program.

**Comment by ghost** [comment]
I wonder if this could be related to the fix that was made to keep the width of the style editor and level strip when OT is maximized.

**Comment by Agnyy** [comment]
This problem appeared a long time ago. I wrote about it. and it is knitted with Style Editor.

https://github.com/opentoonz/opentoonz/issues/2840


https://youtu.be/8cM8OQ5uM5w

**Comment by manongjohn** [comment]
@Agnyy 

This behavior is likely related to Intel HD Graphics Drivers issue that has been affecting a number of OT users in the past few months.  There is a certain version this started happening in and a later version fixes it.

If you have an Intel Graphics display driver, you may need to find and download the latest version. Don't rely on Driver Update as it may not detect it. Need to go onto intel's driver download site and check the version of the driver you have against yours. Manually download and install the driver upgrade it if you are not at the latest.

If you have driver version 26.20.100.7463 or a version of this graphic driver, try downloading and installing version 26.20.100.7584. Here is the direct download link:

https://downloadmirror.intel.com/29274/a08/igfx_win10_100.7584.exe

As you have an NVIDIA graphics card, this may work better for you: Enter the NVIDIA Control Panel and create a 3D Profile for OpenToonz.exe and if given the option, set to use High Performance GPU. You are essentially forcing OT to use your GPU which should fix the issue.

**Comment by MsBlockCat** [comment]
Is there a way for OpenToonz to see the version of the Intel graphics driver being used? If so, maybe we could show a warning when you open OpenToonz on affected versions, and how to get new drivers. I've seen a lot of bug reports on the driver issue!

**Comment by manongjohn** [comment]
I may be wrong here, but I think if OT was configured to use OpenGL properly, we wouldn't be having this issue and possibly some others that are caused by us not utilizing OpenGL like we should. It was tried with 1.2, but that rushed implementation backfired and forced us to back out parts of those changes.  The issues encountered were never revisited.

Having said that, platforms are moving away from OpenGL, I believe, so where does that leave us? We continue to suffer with these problems until we reach a point where we are forced to update to not use OpenGL. In the meantime we will continue piss off existing users and turn off potential new users.

Personally, I'd rather see OT fully adopt OpenGL until it requires something else since nothing has been fully decided upon.

**Comment by MsBlockCat** [comment]
Interesting, I didn't know the Intel graphics issue was caused by that. Is OpenGL dying on Windows? I know it is over time on macOS, and Metal is probably theoretically better to use on Macs, and I also don't know how Linux OpenGL support is doing, but I'm surprised to hear platforms are moving away from it besides macOS

**Comment by manongjohn** [comment]
I am getting a bit frustrated constantly fielding this issue on other OT social sites.

Although we are pegging this as a driver issue, I think the driver is exposing an issue within OT, likely a result of how we are semi using OpenGL? 

It's odd because, as the video @agnyy showed, it only happens when the Style Editor is docked and the Color Wheel tab is active. Why is that??? What is so special about the docked Color Wheel that is causing this?

Some users online have learned to undock the Style Editor to get around the issue. It's probably easier to do that than to get people to upgrade drivers or create graphic profiles but it makes OT look like crap.

All this points to an issue with OT and needs to be addressed properly.

**Comment by MsBlockCat** [comment]
That’s understandable, I’ve seen a lot of people ask about this just in the short time I’ve been here. I agree improving how we use OpenGL is the best way to go forward, at least looking into the next couple years. Knowing Apple I bet they’ll fully remove OpenGL support in favor of Metal, but I’d guess we have a few more years before that happens, and the other OSs would benefit from better OpenGL usage long term regardless since to the extent of my knowledge they don’t have any plans for removing support for it.

Edit: Funny enough, one of my PCs with an i3 processor from around 2013 started showing symptoms of this driver/OpenGL issue! Fortunately I know to update the graphics driver, but this problem seems very prevalent...
